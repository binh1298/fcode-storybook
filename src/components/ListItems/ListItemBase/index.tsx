import {
    ListItem as MaterialListItem,
    ListItemProps as MaterialListItemProps,
} from "@mui/material";

export interface ListItemBaseProps extends MaterialListItemProps {}

const ListItemBase = (props: ListItemBaseProps) => {
    return <MaterialListItem {...props} />;
};

export default ListItemBase;
