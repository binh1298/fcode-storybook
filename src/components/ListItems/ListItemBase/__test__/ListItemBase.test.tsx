import { useTheme, Theme } from "@material-ui/core";

import ListItemBase, { ListItemBaseProps } from "..";

import { render, screen } from "@testing-library/react";

describe("<ListItemBase />", () => {
    let props: ListItemBaseProps;
    let theme: Theme;
    const ListItemBaseTest = (props: ListItemBaseProps) => {
        theme = useTheme();
        return <ListItemBase {...props} />;
    };

    it("should exist children with default variant and color", () => {
        props = {
            children: "Test",
        };
        render(<ListItemBaseTest {...props} />);
        const muiListItem = screen.getByTestId("ListItemBase__root");
        expect(muiListItem).toHaveStyle("justify-content: flex-start;");
        expect(muiListItem).toHaveStyle(`color: ${theme.palette.primary.contrastText};`);
        expect(screen.getByText("Test")).toBeInTheDocument();
    });

    it("should exist children with default variant and custom color", () => {
        props = {
            children: "Test",
            color: "success",
        };
        render(<ListItemBaseTest {...props} />);
        const muiListItem = screen.getByTestId("ListItemBase__root");
        expect(muiListItem).toHaveStyle("justify-content: flex-start;");
        expect(muiListItem).toHaveStyle(`color: ${theme.palette.success.contrastText};`);
        expect(screen.getByText("Test")).toBeInTheDocument();
    });

    it("should exist children with default variant and hightlight color", () => {
        props = {
            children: "Test",
            highlighted: true,
        };
        render(<ListItemBaseTest {...props} />);
        const muiListItem = screen.getByTestId("ListItemBase__root");
        expect(muiListItem).toHaveStyle("justify-content: flex-start;");
        expect(muiListItem).toHaveStyle(`color: ${theme.palette.secondary.main};`);
        expect(screen.getByText("Test")).toBeInTheDocument();
    });

    it("should exist children with default variant and custom hightlight color", () => {
        props = {
            children: "Test",
            highlighted: true,
            hlcolor: "success",
        };
        render(<ListItemBaseTest {...props} />);
        const muiListItem = screen.getByTestId("ListItemBase__root");
        expect(muiListItem).toHaveStyle("justify-content: flex-start;");
        expect(muiListItem).toHaveStyle(`color: ${theme.palette.success.main};`);
        expect(screen.getByText("Test")).toBeInTheDocument();
    });

    it("should exist children with left variant and color", () => {
        props = {
            children: "Test",
            variant: "left",
        };
        render(<ListItemBaseTest {...props} />);
        const muiListItem = screen.getByTestId("ListItemBase__root");
        expect(muiListItem).toHaveStyle("justify-content: flex-start;");
        expect(muiListItem).toHaveStyle(`color: ${theme.palette.primary.contrastText};`);
        expect(screen.getByText("Test")).toBeInTheDocument();
    });

    it("should exist children with left variant and custom color", () => {
        props = {
            children: "Test",
            variant: "left",
            color: "success",
        };
        render(<ListItemBaseTest {...props} />);
        const muiListItem = screen.getByTestId("ListItemBase__root");
        expect(muiListItem).toHaveStyle("justify-content: flex-start;");
        expect(muiListItem).toHaveStyle(`color: ${theme.palette.success.contrastText};`);
        expect(screen.getByText("Test")).toBeInTheDocument();
    });

    it("should exist children with left variant and hightlight color", () => {
        props = {
            children: "Test",
            variant: "left",
            highlighted: true,
        };
        render(<ListItemBaseTest {...props} />);
        const muiListItem = screen.getByTestId("ListItemBase__root");
        expect(muiListItem).toHaveStyle("justify-content: flex-start;");
        expect(muiListItem).toHaveStyle(`color: ${theme.palette.secondary.main};`);
        expect(screen.getByText("Test")).toBeInTheDocument();
    });

    it("should exist children with left variant and custom hightlight color", () => {
        props = {
            children: "Test",
            variant: "left",
            highlighted: true,
            hlcolor: "success",
        };
        render(<ListItemBaseTest {...props} />);
        const muiListItem = screen.getByTestId("ListItemBase__root");
        expect(muiListItem).toHaveStyle("justify-content: flex-start;");
        expect(muiListItem).toHaveStyle(`color: ${theme.palette.success.main};`);
        expect(screen.getByText("Test")).toBeInTheDocument();
    });

    it("should exist children with center variant and color", () => {
        props = {
            children: "Test",
            variant: "center",
        };
        render(<ListItemBaseTest {...props} />);
        const muiListItem = screen.getByTestId("ListItemBase__root");
        expect(muiListItem).toHaveStyle("justify-content: center;");
        expect(muiListItem).toHaveStyle(`color: ${theme.palette.primary.contrastText};`);
        expect(screen.getByText("Test")).toBeInTheDocument();
    });

    it("should exist children with center variant and custom color", () => {
        props = {
            children: "Test",
            variant: "center",
            color: "success",
        };
        render(<ListItemBaseTest {...props} />);
        const muiListItem = screen.getByTestId("ListItemBase__root");
        expect(muiListItem).toHaveStyle("justify-content: center;");
        expect(muiListItem).toHaveStyle(`color: ${theme.palette.success.contrastText};`);
        expect(screen.getByText("Test")).toBeInTheDocument();
    });

    it("should exist children with center variant and hightlight color", () => {
        props = {
            children: "Test",
            variant: "center",
            highlighted: true,
        };
        render(<ListItemBaseTest {...props} />);
        const muiListItem = screen.getByTestId("ListItemBase__root");
        expect(muiListItem).toHaveStyle("justify-content: center;");
        expect(muiListItem).toHaveStyle(`color: ${theme.palette.secondary.main};`);
        expect(screen.getByText("Test")).toBeInTheDocument();
    });

    it("should exist children with center variant and custom hightlight color", () => {
        props = {
            children: "Test",
            variant: "center",
            highlighted: true,
            hlcolor: "success",
        };
        render(<ListItemBaseTest {...props} />);
        const muiListItem = screen.getByTestId("ListItemBase__root");
        expect(muiListItem).toHaveStyle("justify-content: center;");
        expect(muiListItem).toHaveStyle(`color: ${theme.palette.success.main};`);
        expect(screen.getByText("Test")).toBeInTheDocument();
    });

    it("should exist children with right variant and color", () => {
        props = {
            children: "Test",
            variant: "right",
        };
        render(<ListItemBaseTest {...props} />);
        const muiListItem = screen.getByTestId("ListItemBase__root");
        expect(muiListItem).toHaveStyle("justify-content: flex-end;");
        expect(muiListItem).toHaveStyle(`color: ${theme.palette.primary.contrastText};`);
        expect(screen.getByText("Test")).toBeInTheDocument();
    });

    it("should exist children with right variant and custom color", () => {
        props = {
            children: "Test",
            variant: "right",
            color: "success",
        };
        render(<ListItemBaseTest {...props} />);
        const muiListItem = screen.getByTestId("ListItemBase__root");
        expect(muiListItem).toHaveStyle("justify-content: flex-end;");
        expect(muiListItem).toHaveStyle(`color: ${theme.palette.success.contrastText};`);
        expect(screen.getByText("Test")).toBeInTheDocument();
    });

    it("should exist children with right variant and hightlight color", () => {
        props = {
            children: "Test",
            variant: "right",
            highlighted: true,
        };
        render(<ListItemBaseTest {...props} />);
        const muiListItem = screen.getByTestId("ListItemBase__root");
        expect(muiListItem).toHaveStyle("justify-content: flex-end;");
        expect(muiListItem).toHaveStyle(`color: ${theme.palette.secondary.main};`);
        expect(screen.getByText("Test")).toBeInTheDocument();
    });

    it("should exist children with right variant and custom hightlight color", () => {
        props = {
            children: "Test",
            variant: "right",
            highlighted: true,
            hlcolor: "success",
        };
        render(<ListItemBaseTest {...props} />);
        const muiListItem = screen.getByTestId("ListItemBase__root");
        expect(muiListItem).toHaveStyle("justify-content: flex-end;");
        expect(muiListItem).toHaveStyle(`color: ${theme.palette.success.main};`);
        expect(screen.getByText("Test")).toBeInTheDocument();
    });
});
