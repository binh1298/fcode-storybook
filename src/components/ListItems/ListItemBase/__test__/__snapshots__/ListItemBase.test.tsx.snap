// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonBase /> should match snapshot 1`] = `
<div>
  <li
    class="MuiListItem-root MuiListItem-gutters"
  >
    <div>
      Test
    </div>
  </li>
</div>
`;
